src/arkit/ARKitDevice.js

```diff
@@ -26,9 +26,16 @@ export default class ARKitDevice extends PolyfilledXRDevice {
 		// A div prepended to body children that will contain the session layer
 		this._wrapperDiv = document.createElement('div')
 		this._wrapperDiv.setAttribute('class', 'arkit-device-wrapper')
-		document.addEventListener('DOMContentLoaded', ev => {
-			document.body.insertBefore(this._wrapperDiv, document.body.firstChild || null)
-		})
+
+		const insertWrapperDiv = () => {
+			document.body.insertBefore(this._wrapperDiv, document.body.firstChild || null);
+		};
+
+		if (document.body) {
+			insertWrapperDiv();
+		} else {
+			document.addEventListener('DOMContentLoaded', ev => insertWrapperDiv);
+		}
```

Comment:
This change is necessary to support asynchronous polyfill load. If you asynchronously load the polyfill, `codument.body` can be already loaded when the polyfill is loaded. In that case `DOMContentLoaded` event won't be fired. So I synchronously insert the wrapper div here if `document.body` is already ready.

Reply:
Is this the shim, or the full polyfill?  I don't think we'll be doing either:
- the shim will be injected by the app BEFORE the page loads.
- the shim will ask the app to inject the full polyfill only when the page requests WebXR

Reply:
The polyfill.

> the shim will ask the app to inject the full polyfill only when the page requests WebXR

When the page requests WebXR the shim injects(loads) the polyfill. (I called this "on demand polyfill loading" "asynchronous polyfill load".) In this case, the page is already loaded so as I wrote `DOMContentLoaded` event won't be fired so the wrapper div won't be inserted into `document.body` without this change.

In short, these lines were designed only for synchronous polyfill load (meaning loading the polyfill before the page loads) so I needed to update to support the case where the polyfill is loaded after the page is loaded.

Reply:
ok, sounds right!